pwiz_tools/Skyline/Model/DdaSearch/MsFraggerSearchEngine.cs

```diff
@@ -341,7 +341,10 @@ private void FixMSFraggerPin(string cruxInputFilepath, string cruxFixedInputFile
 
         private void EnsureFastaHasDecoys()
         {
-            _progressStatus = _progressStatus.ChangeMessage(string.Format(Resources.EnsureFastaHasDecoys_Detecting_decoy_prefix_in__0__, Path.GetFileName(fastaFilepath)));
+            Assume.IsNotNull(_fastaFilepath);

```

Comment:
Consider putting this after the string.Empty assignment and then check Assume.IsFalse(string.IsNullOrWhitespace()) since it certainly shouldn't be empty (or even a non-existing filepath) coming in to this function.

Reply:
Sure